codegen/validation.go

```diff
@@ -325,8 +349,8 @@ func recurseAttribute(att *expr.AttributeExpr, attCtx *AttributeContext, nat *ex
 		if hasValidations {
 			var buf bytes.Buffer
 			tgt := fmt.Sprintf("%s.%s", target, attCtx.Scope.Field(nat.Attribute, nat.Name, true))
-			if expr.IsArray(nat.Attribute.Type) {
```

Comment:
No.. Before, we generated the validation code for a result type collection inline instead of creating a new `ValidateXXX` function with the validation code. By this change, we will be generating validation code for aliased primitive types inline and create a new `ValidateXXX` function for result type collection and other user types (aliased or unaliased). I felt it made managing this code a little easier.

Reply:
Here is the effect of this change - https://github.com/goadesign/goa/blob/4f9dd318c61be04cd8dfb3fbbdd4e46c1b34e78a/codegen/testdata/validation_code.go#L383-L387
Notice how it generates this function `ValidateResultCollection` for validating result type collection
```go
        if err2 := ValidateResultCollection(target.Collection); err2 != nil {
		err = goa.MergeErrors(err, err2)
	}
```
Without this change, it would have generated a `for` loop and ran the validation for each item in the collection. Now, the iteration is done inside the generated function. I feel this is more consistent because now we know a new `ValidateXXX` function will be generated for all user types except for aliased primitive types.

Reply:
OK makes sense, the initial intent was to avoid generating extra functions but you make a good point that it is more consistent this way.